From a772afe0e330499261d5d503b1cfc758c9b15fe6 Mon Sep 17 00:00:00 2001
From: "James T. Lee" <james@james.tl>
Date: Mon, 22 Jun 2020 15:47:57 -0400
Subject: [PATCH] Revert "gbm/dri: Propagate queryDmaBufModifiers return value"

This reverts commit e58509cdeccc85adcf9127c6d1a462a73e7d5068.  It breaks
xwayland on panfrost with the message "Modifiers specified, but DRI is
too old."
---
 src/gbm/backends/dri/gbm_dri.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/gbm/backends/dri/gbm_dri.c b/src/gbm/backends/dri/gbm_dri.c
index bc68f66766e..8fe6a0265fb 100644
--- a/src/gbm/backends/dri/gbm_dri.c
+++ b/src/gbm/backends/dri/gbm_dri.c
@@ -618,12 +618,14 @@ gbm_dri_is_format_supported(struct gbm_device *gbm,
       }
    }
 
-   /* This returns false if the format isn't supported */
+   /* Check if the driver returns any modifiers for this format; since linear
+    * is counted as a modifier, we will have at least one modifier for any
+    * supported format. */
    if (!dri->image->queryDmaBufModifiers(dri->screen, format, 0, NULL, NULL,
                                          &count))
       return 0;
 
-   return 1;
+   return (count > 0);
 }
 
 static int
-- 
2.26.2

